test/integ/test_put_get_medium.py

```diff
@@ -30,41 +31,52 @@
 except ImportError:
     CONNECTION_PARAMETERS_ADMIN = {}
 
-
 THIS_DIR = os.path.dirname(os.path.realpath(__file__))
 logger = getLogger(__name__)
 
 
-def test_put_copy0(conn_cnx, db_parameters):
+@pytest.fixture()
+def file_src(request) -> Tuple[str, int, IO[bytes]]:
+    file_name = request.param
+    data_file = os.path.join(THIS_DIR, "../data", file_name)
+    file_size = os.stat(data_file).st_size
+    stream = open(data_file, 'rb')
+    yield data_file, file_size, stream
+    stream.close()
+
+
+@pytest.mark.parametrize("from_stream", [False, pytest.param(True, marks=pytest.mark.skipolddriver)])
+@pytest.mark.parametrize("file_src", ['put_get_1.txt'], indirect=['file_src'])
+def test_put_copy0(conn_cnx, db_parameters, from_stream, file_src):
     """Puts and Copies a file."""
-    data_file = os.path.join(THIS_DIR, "../data", "put_get_1.txt")
+    file_path, _, file_stream = file_src
+    kwargs = {'_put_callback': SnowflakeS3ProgressPercentage,
+              '_get_callback': SnowflakeS3ProgressPercentage,
+              '_put_azure_callback': SnowflakeAzureProgressPercentage,
+              '_get_azure_callback': SnowflakeAzureProgressPercentage,
+              'file_stream': file_stream}
+
+    if not from_stream:
+        kwargs.pop('file_stream', None)
 
     def run(cnx, sql):
         sql = sql.format(
-            file=data_file.replace('\\', '\\\\'),
+            file=file_path.replace('\\', '\\\\'),
             name=db_parameters['name'])
```

Comment:
I think this test is already using randomized name generator for `table_name` and such, I just forgot to delete `create or replace`, it will work with just ` create`